man/adoc/bpftrace.adoc

```diff
@@ -3556,6 +3556,16 @@ The possible options are:
 - `warn` - print a warning but continue execution
 - `ignore` - silently ignore missing probes
 
+=== on_stack_limit
+
+Default: 32
+
+The maximum size (in bytes) of individual objects that will be stored on the BPF stack. If they are larger than this threshold they will be stored in
+pre-allocated memory.
+
+This threshold is necessary because the BPF stack is limited to 512 bytes and large objects make it more likely for the stack to run out of space. Instead,
+bpftrace can store these large objects in read-write global variables. However, this involves pre-allocating memory so it may be less memory efficient.
+
```

Comment:
nit: slight re-wording

```suggestion
This exists because the BPF stack is limited to 512 bytes and large objects make it more likely that we'll run out of space. bpftrace can store objects that are larger than the `on_stack_limit` in pre-allocated memory to prevent this stack error. However, storing in pre-allocated memory may be less memory efficient. Lower this default number if you are still seeing a stack memory error or increase it if you're worried about memory consumption.`

```